dynawo/cmake/Support-LibXml2.cmake

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+# See AUTHORS.txt
+# All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+
+cmake_minimum_required( VERSION 3.12)
+
+if ( NOT CMAKE_SCRIPT_MODE_FILE )
+    project( "Libxml2 support for Dynawo")
+endif ()
+
+include( $ENV{DYNAWO_HOME}/dynawo/cmake/cpu_count.cmake)
+if ( NOT DEFINED CPU_COUNT )
+    message( FATAL_ERROR "cpu_count.cmake: file not found.")
+endif ()
+
+set( paquet_name        "libxml2")
+set( paquet_finder      "LibXml2")
+set( paquet_install_dir "${CMAKE_INSTALL_PREFIX}/libxml2")
+
+string( TOLOWER ${paquet_name} paquet_lowername)
+string( TOUPPER ${paquet_name} paquet_uppername)
+
+unset( CMAKE_MODULE_PATH)
+list( APPEND CMAKE_MODULE_PATH "$ENV{DYNAWO_HOME}/dynawo/cmake")
+if ( IS_DIRECTORY "$ENV{DYNAWO_HOME}/dynawo/3rdParty/${paquet_name}")
+    list( APPEND CMAKE_MODULE_PATH "$ENV{DYNAWO_HOME}/dynawo/3rdParty/${paquet_name}")
+endif ()
+
+set(RequiredVersion 2.9)
+
+set( CMAKE_PREFIX_PATH "${CMAKE_INSTALL_PREFIX}/libxml2")
+find_package( ${paquet_finder} ${RequiredVersion} QUIET)
+if ( ${paquet_finder}_FOUND )
+
+    add_custom_target( ${paquet_name})
+    set( ${paquet_uppername}_HOME ${paquet_install_dir})
+    message( STATUS "found LibXML2 lib and includes : ${LIBXML2_VERSION_STRING} in ${LIBXML2_HOME}")
+
+else ()
+    include( ExternalProject)
+    ExternalProject_Add(
+                            ${paquet_name}
+        INSTALL_DIR         ${paquet_install_dir}
+
+        GIT_REPOSITORY      https://gitlab.gnome.org/GNOME/libxml2.git
+        GIT_TAG             master
```

Comment:
should be ${RequiredVersion}

Reply:
I require at least one version (2.9), but if it is needed, I download latest one. If you want I can force a required version, that may be more safe. In that case, which one ? powsybl does not set one. The libxml2 library is already in Linux. A safer solution could be to "dnf install libxml2-devel" or "apt install libxml2-dev". Doing so we may have differences across systems...

Reply:
We need to download a release and not a master that is not stable. Powsybl requires 2.9 as a minimal version so that is the version we should install. The solution dnf or apt is only there for linux and will be the prefered way to install it on those os but for macos and windows we will use the code you juste developed

Reply:
You are right. But powsybl does not set any requirement for LibXML2 version. I did it on my own, without any reason. Choosing a version we must say something like 2.9.4, but it will be our choice, arbitrary. My own opinion is to say LibXML2 is inside Gnome, so we must install the package libxml2-dev or devel from apt or dnf. It is as other libraries we use, like libblas, libaplack, libarchive or libexpat (examples). In that way we won't have trouble to garranty we use the same library powsybl does. That is an important choice to make since the actual code uses 'git clone': easy to deploy, but instable, I agree. I have 3 samples of Support-LibXML2.cmake that uses different way to install that part... My opinion is that following powsybl way to solve the question will be our best choice

Reply:
On windows system, and MacIntosh, we will do exactly as it is for other packages like libblas... and the new libIIDM. There must be an indication on installing LibXML2 in powsybl on Windows (?)